symengine/ruby/ext/symengine/symengine.c

```diff
@@ -22,4 +24,11 @@ void Init_symengine() {
     rb_define_method(c_basic, "/", cbasic_div, 1);
     rb_define_method(c_basic, "**", cbasic_pow, 1);
     rb_define_method(c_basic, "-@", cbasic_neg, 0);
+    rb_define_method(c_basic, "to_s", cbasic_to_str, 0);
+    rb_define_method(c_basic, "expand", cbasic_expand, 0);
+
+    //Symbol class
+    c_symbol = rb_define_class_under(m_symengine, "Symbol", c_basic);
+    rb_define_alloc_func(c_symbol, csymbol_alloc);
```

Comment:
But I don't think that's how it works with the Ruby C API. The function is just given to allocate such an instance and return the object. It's just the C API counterpart of ".allocate" method in ruby. I decided to check it out and added a `printf` statement in `cbasic_alloc`.
``` c
VALUE cbasic_alloc(VALUE klass){
    VALUE obj;
    basic_struct *struct_ptr;

    obj = Data_Make_Struct(klass, basic_struct, NULL, cbasic_free, struct_ptr);

    struct_ptr->data = NULL;
    printf("basic object allocated\n");

    return obj;
}
```
This is the output of the test files
``` bash
SymEngine
  SymEngine::Basic
    .new
      with no arguments
basic object allocated
        returns a Basic object

SymEngine
  SymEngine::Symbol
    .new
      with an integer as an argument
basic object allocated
        returns a Symbol object
    binary operations
      #+
        with another Symbol object as argument
basic object allocated
basic object allocated
          returns a Symbol object that is result of
                        self added to the argument
      #-
        with another Symbol object as argument
basic object allocated
basic object allocated
          returns a Symbol object that is result of
                        argument subtracted from self
      #*
        with another Symbol object as argument
basic object allocated
basic object allocated
          returns a Symbol object that is result of
                        self multiplied by the argument
      #/
        with another Symbol object as argument
basic object allocated
basic object allocated
          returns a Symbol object that is result of
                        self divided by the argument
      #**
        with another Symbol object as argument
basic object allocated
basic object allocated
          returns a Symbol object that is result of
                        self raised to the power of argument
    unary operations
      #-@
        doesn't take any argument
basic object allocated
          returns the negation of self
    #expand
      with nothing as argument
basic object allocated
basic object allocated
basic object allocated
        returns the expanded form

Finished in 0.00452 seconds (files took 0.09732 seconds to load)
9 examples, 0 failures
```
This shows they are not being called twice.